power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -665,8 +672,13 @@ class TapPositionOptimizerImpl<std::tuple<TransformerTypes...>, StateCalculator,
                 if (tap_changed) {
                     break;
                 }
+                iterations_per_rank[++rank_index] = 0;
             }
             if (tap_changed) {
+                if (++iterations_per_rank[rank_index] > 2 * regulator_order[rank_index][0].transformer.tap_range()) {
```

Comment:
should be (pseudocode):
```
max_iters[rank_k] = max( tap_range() for transformer in regulator_order[rank_idx] )
```
because trafo 1 in the same rank may have a larger tap range than trafo 0.

probably a good idea to pre-calculate (maybe even in a separate function)

Reply:
Conclusion: maximum tap range should be taken